### PR TITLE
TX-2718 | insertion of custom headers into event map was missing

### PR DESCRIPTION
**Pull request by vivekyadav1998**
issue found= while preparing the event map, we just skip the insertion of custom headers into event map, this cause issue and we were not able to receive custom headers values in park event from freeswitch.

fix provide = we add some custom handing which ensure to insert all the provided custom headers in the form of string into the event map.

### CODE REVIEW

File: eventsocket/eventsocket.go [hunk 1]
```diff
@@ -305,6 +305,8 @@ func (h *Connection) ReadEvent() ([]string, error) {
 // It's used after parsing plain text event headers, but not JSON.
 func copyHeaders(src *textproto.MIMEHeader, dst []string, decode bool) {
 	var err error
+	var customHeaderValues string
```
Comment (anshumanr) [inline eventsocket/eventsocket.go:307]: so I would organize the code in this way -
```
if decode {
//both conditions here
} else {
//all non-decode conditons here
}
```

File: eventsocket/constants.go [hunk 2]
```diff
@@ -162,3 +162,4 @@ var MapKeyIndex = map[string]int{
 }
 
 const FsEventMapSize = 100
+const customHeaderPrefix = "Variable_sip_h_x"
```
Comment (anshumanr) [inline eventsocket/constants.go]: is it possible to add this to the above block, maybe just to keep it consistent.

File: eventsocket/eventsocket.go (hunk 1 above)
Comment (vivekyadav1998) [reply eventsocket/eventsocket.go:307]: did changes acc. to your suggestions.

File: eventsocket/eventsocket.go [hunk 3]
```diff
@@ -305,17 +305,36 @@ func (h *Connection) ReadEvent() ([]string, error) {
...
 			}
-		} else if MapKeyIndex[k] != 0 {
-			dst[MapKeyIndex[k]] = v[0]
 		}
 	}
+
+	if customHeaderValues != "" {
+		dst[CustomHeaders] = customHeaderValues
+	}
 }
```
Comment (anshumanr) [inline eventsocket/eventsocket.go]: After you made those changes, i feel this is even cleaner, if it works
```
func copyHeaders(src *textproto.MIMEHeader, dst []string, decode bool) {
	var customHeaderValues string
	clen := len(customHeaderPrefix)
	for k, v := range *src {
		k = capitalize(k)
		if decode {
			escapedHeader, err := url.QueryUnescape(v[0])
			if err == nil {
				v[0] = escapedHeader
			}
		}

		if MapKeyIndex[k] != 0 {
			dst[MapKeyIndex[k]] = v[0]
		} else if len(k) >= clen && k[0:clen] == customHeaderPrefix {
			customHeaderValues = customHeaderValues + k + ":" + v[0] + "|"
		}
	}

	if customHeaderValues != "" {
		dst[CustomHeaders] = customHeaderValues
	}
}
```

Comment (vivekyadav1998) [reply eventsocket/eventsocket.go]: yes, i will execute this code and will update here, it is working or  not.

**Review by anshumanr — APPROVED** [review]

**Review by lazyboson — APPROVED** [review]

File: eventsocket/eventsocket.go (hunk 3 above)
Comment (vivekyadav1998) [reply eventsocket/eventsocket.go]: yes it is working

**Pull request merged**